### ISSUE TITLE
v1.6 defdelegate does not generate default @doc anymore

### ISSUE THREAD

**Issue by ianrumford** [elixir-lang/elixir#7304]
### Environment

Erlang/OTP 20 [erts-9.2] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:10] [hipe] [kernel-poll:false]

Elixir 1.6.1 (compiled with OTP 20)

### Current behavior

The defdelegate macro no longer generates a default @doc attribute.

I think this is the result of the 8ef3a25 commit on 6Jan18 to address a problem with multi-clause functions that have a delegate as the last clause.

I understand that's an issue that needs to be addressed but I believe the fix sacrifices the common usage case for defdelegate where the auto-generated @doc is exactly what you want.

Any older packages expecting the default behaviour but compiled and published to Hex against 1.6 will loose the @doc and break the documentation leaving an "orphan" function.

I could not find anything in the release notes about this change.

### Expected behavior

I would like to see the 1.5 behaviour reinstated for the common use case.

**Comment by josevalim** [comment]
I will update the changelog because I don't see a mechanism that doesn't break users expectations somehow. At least not generating something is less surprising than overriding something. Thanks for the report!

**Comment by ianrumford** [comment]
Silently not generating something which was generated *is* more surprising!

**Comment by josevalim** [comment]
Temporarily yes, especially when we forgot to add it to the CHANGELOG. :)
But not in the long term.
-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Comment by ianrumford** [comment]
Elixir has some very good guidance on when and how to deprecate functions.  It is regrettable that that discipline wasn't used in this case.  

TL;DR Don't pull the rug out from under, especially silently.
